src/farmer.rs

```diff
@@ -22,6 +16,44 @@ use tokio::sync::{oneshot, watch, Mutex};
 
 use crate::{Node, PublicKey};
 
+/// Description of the cache
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct CacheDescription {
+    /// Path of the plot
+    pub directory: PathBuf,
+    /// Space which you want to pledge
+    pub space_pledged: ByteSize,
+}
```

Comment:
these comments are confusing imo. Path of the plot and space I want to pledge sounds like the exact same thing with `PlotDescription`. However, we both ned `Cache` and `Plot` in order to start farming I guess, so they should be different. 
These comments may highlight those differences maybe?

Reply:
I mean, `CacheDescription` is description of cache, while `PlotDescription` is description of plot. I think their names speak clearly about them.

Reply:
`/// path of the plot` -> line 23

Reply:
So I'm pledging say 10GB space, denoted with PlotDescription,
and additional 1GB space, denoted with CacheDescription,
in total 11GB ?

I thought I'm only pledging space with my plot, not with DSN configuration. Maybe my understanding is wrong here

Reply:
> So I'm pledging say 10GB space, denoted with PlotDescription,
>  and additional 1GB space, denoted with CacheDescription,
>  in total I'll earn rewards from 11GB ?

Nope, here 1Gb is just an overhead for DSN, you won't get any rewards now for it.

Reply:
Yes I thought so. Hence my first comment.
`CacheDescription` does not define the plot, nor the space pledged.
So far, `Pledge` in terms of `farming`, is always used for the space that you will earn rewards from. We were not pledging space to node, although node was occupying space back then.

The wording should not be `plot` and `pledge` imo. I find it very confusing

Reply:
And why should someone give more space to cache overhead? 
I mean, for node, it was archiving the whole history but was not contributing to rewards, so increasing the size for node would archive more, contributing to the decentralization without any profit to the farmer.

For increasing the `farm` size, the farmer would earn more rewards.

For increasing the cache size of dsn, why would anyone do that? Maybe comments can explain this as well (faster sync time, archiving more, etc.)